### PR TITLE
Implptr Get function

### PR DESCRIPTION
**Pull request by nkoenig**
# 🎉 New feature

## Summary

Adds a get function to the ImplPtr class so that you can do things like:

```
class Implementation
{
  public: void Run();
  public: std::thread runThread;
};

class MainClass
{
  public: void Load()
  {
    this->dataPtr->runThread = std::thread(&Implementation::Run, this->dataPtr.Get());
  }
}; 
```
## Test it

Run the tests.

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [X] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [X] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

**Comment by chapulina** [comment]
Can this target `ign-utils1`?

File: test/integration/implptr/ImplPtr_TEST.cc [hunk 1]
```diff
@@ -111,6 +120,15 @@ TEST(UniqueImplPtr, MoveConstruct)
   EXPECT_EQ(other.GetString(), moved.GetString());
 }
 
+/////////////////////////////////////////////////
+TEST(UniqueImplPtr, GetAccessor)
+{
+  MovableObject object(28, "golden_string");
```
[the remark this answers is not in the document]
Comment (chapulina) [reply test/integration/implptr/ImplPtr_TEST.cc:126]: ping @nkoenig

File: test/integration/implptr/implptr_test_classes.cc [hunk 2]
```diff
@@ -17,13 +17,20 @@
 
 #include "implptr_test_classes.hh"
 
+#include <future>
```
[the remark this answers is not in the document]
Comment (nkoenig) [reply test/integration/implptr/implptr_test_classes.cc]: Removed in 6419a1b

[the remark this answers is not in the document]
Comment (nkoenig) [reply test/integration/implptr/implptr_test_classes.cc]: Done, I think, in 6419a1b

File: test/integration/implptr/ImplPtr_TEST.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (nkoenig) [reply test/integration/implptr/ImplPtr_TEST.cc:136]: Done, I think, in 6419a1b

**Review by azeey — APPROVED** [review]

**Pull request merged**